### PR TITLE
1501 task2 basic web api crud

### PR DESCRIPTION
**Pull request by mjunaid-cnc**

### CODE REVIEW

File: Task2-BasicWebApiCRUD/Controllers/WeatherForecastController.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+using Microsoft.AspNetCore.Mvc;
+
```
Comment (Abdul-rehman-butt) [inline Task2-BasicWebApiCRUD/Controllers/WeatherForecastController.cs]: Remove this file

File: Task2-BasicWebApiCRUD/WeatherForecast.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+namespace Task2_BasicWebApiCRUD
+{
+    public class WeatherForecast
+    {
+        public DateTime Date { get; set; }
+
+        public int TemperatureC { get; set; }
```
Comment (Abdul-rehman-butt) [inline Task2-BasicWebApiCRUD/WeatherForecast.cs]: Remove this file

File: Todo.Application/Class1.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Todo.Application
```
Comment (Abdul-rehman-butt) [inline Todo.Application/Class1.cs]: Remove unused files

File: Task2-BasicWebApiCRUD/Controllers/ToDoController.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            _todoService = todoService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> AddItem(ToDoModel model)
+        {
+            try
+            {
+                var addedItem = await _todoService.AddItem(model);
+                return Ok(addedItem);
```
Comment (Abdul-rehman-butt) [inline Task2-BasicWebApiCRUD/Controllers/ToDoController.cs]: Implement appropriate error handling for cases where the requested resource is not found or there is a validation error.
Provide appropriate HTTP status codes and responses for each endpoint.

Where are we handling this?

File: Task2-BasicWebApiCRUD/Controllers/WeatherForecastController.cs (hunk 1 above)
Comment (mjunaid-cnc) [reply Task2-BasicWebApiCRUD/Controllers/WeatherForecastController.cs]: Done

File: Task2-BasicWebApiCRUD/WeatherForecast.cs (hunk 2 above)
Comment (mjunaid-cnc) [reply Task2-BasicWebApiCRUD/WeatherForecast.cs]: Done

File: Todo.Application/Class1.cs (hunk 3 above)
Comment (mjunaid-cnc) [reply Todo.Application/Class1.cs]: Done

File: Task2-BasicWebApiCRUD/Controllers/ToDoController.cs (hunk 4 above)
Comment (mjunaid-cnc) [reply Task2-BasicWebApiCRUD/Controllers/ToDoController.cs]: Screenshot 2023-10-13 161748

Please see the above screenshot for error handling for the case where the resource is not found.

**Pull request merged**